src/orders/components/OrderPayment/utils.ts

```diff
@@ -11,24 +9,23 @@ export const extractOrderGiftCardUsedAmount = (
     return undefined;
   }
 
-  const { id, giftCards } = order;
+  const { giftCards } = order;
 
-  const orderRelatedEvent = giftCards.reduce((resultEvent, { events }) => {
-    if (resultEvent) {
-      return resultEvent;
-    }
+  const usedInOrderEvents = compact(
+    giftCards.map(({ events }) =>
+      events.find(({ type }) => type === GiftCardEventsEnum.USED_IN_ORDER)
```

Comment:
Do we need only first event here?

Reply:
Ah yes, id got lost in the refactor, adding